### PR TITLE
Issue 5951/check untyped defs apache

### PR DESCRIPTION
**Pull request by dschlessman**
* Add types where `tox -e mypy` reported errors
* Addressed mypy error where logger.fatal function not found: logger.critical used instead.
* Removed unused conf argument in get_apache_configurator due to mypy error, "error: Cannot assign to a method”, in line `config.conf = conf`

### CODE REVIEW

File: certbot-apache/certbot_apache/configurator.py [hunk 1]
```diff
@@ -150,14 +152,14 @@ def __init__(self, *args, **kwargs):
         super(ApacheConfigurator, self).__init__(*args, **kwargs)
 
         # Add name_server association dict
-        self.assoc = dict()
+        self.assoc = dict()  # type: Dict[str, obj.VirtualHost]
         # Outstanding challenges
-        self._chall_out = set()
+        self._chall_out = set()  # type: Set[Tuple[KeyAuthorizationAnnotatedChallenge, ...]]
```
Comment (dschlessman) [inline certbot-apache/certbot_apache/configurator.py]: Due to enumerate(iterable) used later on self._chall_out in another function, it appears that a list may be valid as well. In this case the type could be Set[Union[List[KeyAuthorizationAnnotatedChallenge], Tuple[KeyAuthorizationAnnotatedChallenge, ...]]]. However, the automated tests use a tuple. And the aforementioned type annotation is about 109 characters by itself, which leads to issues passing pylint. I don't think there is a valid way to break up that type annotation to pass pylint and be a valid type annotation.

File: certbot-apache/certbot_apache/tests/util.py [hunk 2]
```diff
@@ -133,10 +132,6 @@ def mock_os_constant(key, vhost_path=vhost_path):
                         config_class = configurator.ApacheConfigurator
                     config = config_class(config=mock_le_config, name="apache",
                         version=version)
-                    # This allows testing scripts to set it a bit more
-                    # quickly
-                    if conf is not None:
-                        config.conf = conf  # pragma: no cover
```
Comment (dschlessman) [inline certbot-apache/certbot_apache/tests/util.py]: This change could use extra attention.

**Comment by bmw** [comment]
Sorry for the cryptic and unexpected test failures here. What's going on is the different pieces packages in this repo are not in required lockstep. For example, you can have a one version of `certbot` and another version of `certbot-apache`. We have tests to make sure that the minimum version we say we require is correct which is failing here because `acme.magic_typing` is new for us.

To fix this, you should modify `certbot-apache/setup.py` to require `acme>0.24.0` and the `acme` line in `certbot-apache/local-oldest-requirements.txt` to `-e acme[dev]`. You can take my word on the latter or I can explain to you how this works if you're curious.

File: certbot-apache/certbot_apache/configurator.py (hunk 1 above)
Comment (dschlessman) [reply certbot-apache/certbot_apache/configurator.py]: You're right. Thank you.

File: certbot-apache/certbot_apache/tests/configurator_test.py [hunk 3]
```diff
@@ -348,19 +348,16 @@ def mock_find_dir(directive, argument, _):
         for a in mock_add.call_args_list:
             if a[0][1] == "Include" and a[0][2] == self.config.mod_ssl_conf:
                 tried_to_add = True
-        # Include should be added, find_dir is not patched, and returns falsy
+        # Include should be added, find_dir is not patched, and returns false
```
[the remark this answers is not in the document]
Comment (dschlessman) [reply certbot-apache/certbot_apache/tests/configurator_test.py]: Oh! Thanks!

**Pull request merged**